### ISSUE TITLE
build(webpack): implement babel-polyfill for browser compatibility

### ISSUE THREAD

**Issue by kevinwhoffman** [WordImpress/Give#3258]
## User Story

As a Give user with an older browser such as IE 10/11, I want Give to work the same as it does in modern browsers so that my experience is not affected by the developer's choice to use modern JS.

## Acceptance Criteria

- [ ] Give core and any add-on using ES6 features has been tested in legacy browsers (especially IE 10+).
- [ ] Any features that result in errors have polyfills to prevent those errors.
- [ ] The polyfills that are added to Give core can be leveraged by any Give add-on so that we don't unnecessarily bloat each add-on with multiple copies of the polyfills.

## Possible Solution

@emgk and I were able to implement [`babel-polyfill`](https://babeljs.io/docs/usage/polyfill/) in the WooCommerce add-on. It worked as expected to polyfill `Array.from()`, however the total file size of `babel-polyfill.js` was 89 KB minimized.

We should explore alternatives that allow us to only include the polyfills we need, such as: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from#Polyfill

**Comment by kevinwhoffman** [comment]
I'm documenting the process that @emgk and I used so we can recreate it in Give core.

## Instructions for Implementing `babel-polyfill`

1. Install `babel-polyfill`.

`npm install --save babel-polyfill`

2. Add `babel-polyfill` to the top of the entry array in `webpack-config.js` like so:

```js
entry: {
    'babel-polyfill': 'babel-polyfill',
    'admin-main': [
        './assets/src/js/admin-main.js',
        './assets/src/css/admin-main.scss'
    ],
    'public-main': [
        './assets/src/js/public-main.js',
        './assets/src/css/public-main.scss'
    ]
},
```

3. Run `npm run dev` to compile the JS. You should see a `babel-polyfill.js` file in your `dist/js` directory.

4. Enqueue this `babel-polyfill.js` file before all other plugin scripts.

**Comment by kevinwhoffman** [comment]
I discovered that `babel-polyfill` uses [`core-js`](https://github.com/WordImpress/Give/issues/3258) under the hood, which _does_ allow you to import only the polyfills you need for specific features.

See: https://github.com/zloirock/core-js#commonjs

**Comment by mehul0810** [comment]
I've tried implementing `babel-polyfill` with Stripe while working on the webpack issue and discussed with @emgk and test the Stripe CC fields not loading on IE11.

## Points we think to be implemented
1. We should load add `babel-polyfill` only in Give Core so that add-ons can leverage it.
2. We need `babel-polyfill` to support IE browsers so don't load it by default, instead, bundle it with Give Core and provide a snippet to customers who need to support older IE versions as Edge 16+ supports ES6 (i.e. https://caniuse.com/#search=ES6) so that using the snippet they can load the polyfill script only on IE browsers. This will only add polyfill script to IE browsers so modern browsers are not highly impacted with unnecessary scripts.
3. Even after adding `babel-polyfill`, Custom Events are not supported in IE browsers. So, there is an additional script which needs to be added in Give Core. Check PolyFills section here https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent

**Comment by kevinwhoffman** [comment]
It's not in our users best interests to require a snippet in order for Give to function in any version of Internet Explorer. Site admins (who are more likely to have modern browsers) are probably not the ones who would notice the breakage anyways; it would be a donor attempting to complete a donation in IE only to find a form is broken. They'll get frustrated and leave without ever reporting it.

We have to remember we're creating this issue ourselves as the result of a development choice, so it should be handled seamlessly without a user or admin ever knowing the difference.

## Recommended Solution

- Include a polyfills script in Give core so that all add-ons can leverage it.
- Be smart about which polyfills we are including so that we're only polyfilling features we are using.
- Use feature detection (potentially through [Modernizr](https://modernizr.com/)) to detect if ES6 features are not supported and load the polyfill script for those browsers. Ultimately this is about feature support, not specifically one browser or another.

**Comment by kevinwhoffman** [comment]
Closing. This was addressed in #3433.
